From 434fffdc43f1bc0efefcf163172e0626723b660f Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Tue, 14 Mar 2023 21:32:36 +0100
Subject: [PATCH] Apply CMAKE_ARGS if set

---
 setup.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/setup.py b/setup.py
index 02ad53f..08e7b0d 100644
--- a/setup.py
+++ b/setup.py
@@ -4,6 +4,7 @@
 import distutils.dir_util
 import os
 import platform
+import shlex
 import sys
 from codecs import open
 from distutils.command.clean import clean as _clean
@@ -88,6 +89,8 @@ class BuildPyLibVWBindingsModule(_build_ext):
             "-DVW_ZLIB_SYS_DEP=On",
             "-DVW_EIGEN_SYS_DEP=On",
         ]
+        if "CMAKE_ARGS" in os.environ:
+            cmake_args += shlex.split(os.environ["CMAKE_ARGS"])
 
         # This doesn't work as expected for Python3.6 and 3.7 on Windows.
         # See bug: https://bugs.python.org/issue39825
-- 
2.37.1 (Apple Git-137.1)

