From ea1309bad64a90b45140e051d85f3519f81d107c Mon Sep 17 00:00:00 2001
From: ngam <67342040+ngam@users.noreply.github.com>
Date: Fri, 28 Jan 2022 23:09:05 -0500
Subject: [PATCH 1/2] extension setting

---
 python/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 7e8884ecd..4b174a766 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -50,7 +50,7 @@ if (NOT WIN32)
   # Windows links dynamically
   target_compile_definitions(pylibvw PUBLIC BOOST_PYTHON_STATIC_LIB)
   # OSX builds a .dylib and then attempts to find a .so. Just build a .so file and things work
-  set_target_properties(pylibvw PROPERTIES SUFFIX ".so")
+  # set_target_properties(pylibvw PROPERTIES SUFFIX ".so")
 else()
   # Force Windows to build with a .pyd extension
   set_target_properties(pylibvw PROPERTIES SUFFIX ".pyd")
-- 
2.34.1

