diff --git a/setup.py b/setup.py
index c05b4ace3..a02122520 100644
--- a/setup.py
+++ b/setup.py
@@ -82,6 +82,12 @@ class BuildPyLibVWBindingsModule(_build_ext):
             "-DBUILD_PYTHON=On",
             "-DBUILD_TESTING=Off",
             "-DWARNINGS=Off",
+            "-DRAPIDJSON_SYS_DEP=On",
+            "-DFMT_SYS_DEP=On",
+            "-DSPDLOG_SYS_DEP=On",
+            "-DVW_BOOST_MATH_SYS_DEP=On",
+            "-DVW_ZLIB_SYS_DEP=On",
+            "-DVW_EIGEN_SYS_DEP=On",
         ]

         # This doesn't work as expected for Python3.6 and 3.7 on Windows.
