--- CMakeLists.txt.orig	2020-10-06 13:58:25.000000000 +0200
+++ CMakeLists.txt	2020-10-06 13:58:56.000000000 +0200
@@ -90,7 +90,7 @@
 
 # Add -ffast-math for speed, remove for testability.
 # no-stack-check is added to mitigate stack alignment issue on Catalina where there is a bug with aligning stack-check instructions, and stack-check became default option
-set(linux_release_config -O3 -fno-strict-aliasing -msse2 -mfpmath=sse -fno-stack-check)
+set(linux_release_config -O3 -fno-strict-aliasing -fno-stack-check)
 set(linux_debug_config -g -O0 -fno-stack-check)
 
 if((NOT PROFILE) AND (NOT GCOV))
