diff --git a/cmake/VWFlags.cmake b/cmake/VWFlags.cmake
index 4d0b151ea..e0bd30964 100644
--- a/cmake/VWFlags.cmake
+++ b/cmake/VWFlags.cmake
@@ -12,11 +12,11 @@ if(LTO)
   set(VW_LINUX_FLAGS ${VW_LINUX_FLAGS} -flto=thin)
 endif()

-if("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
-  if(NOT "arm64" STREQUAL "${CMAKE_OSX_ARCHITECTURES}")
-    set(LINUX_X86_64_OPT_FLAGS -msse2 -mfpmath=sse)
-  endif()
-endif()
+#if("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
+#  if(NOT "arm64" STREQUAL "${CMAKE_OSX_ARCHITECTURES}")
+#    set(LINUX_X86_64_OPT_FLAGS -msse2 -mfpmath=sse)
+#  endif()
+#endif()

 # Add -ffast-math for speed, remove for testability.
 # no-stack-check is added to mitigate stack alignment issue on Catalina where there is a bug with aligning stack-check instructions, and stack-check became default option
